### PR TITLE
Dag Scheduler changes 

### PR DESCRIPTION
**Pull request by Jeyaprakash-NK**

### CODE REVIEW

File: src/scheduler/schedulerServices.tsx [hunk 1]
```diff
@@ -0,0 +1,669 @@
...
+        } else {
+          toast.success(
+            `Job scheduler successfully created`,
+            toastifyCustomStyle
+          );
+        }
+        setCreatingScheduler(false);
+        setCreateCompleted(true);
+      }
+      console.log(data);
```
Comment (harsha-accenture) [inline src/scheduler/schedulerServices.tsx:232]: @Jeyaprakash-NK Remove the console logs

File: style/notebookScheduler.css [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+  margin: 10px;
+  border-bottom: 1px solid var(--jp-layout-color3);
+}
+
+.accordion-row-data-expand-logo {
+  width: 30px;
+}
+
+.dag-runs-table-data-state-success {
+  background-color: #34a853;
```
Comment (harsha-accenture) [inline style/notebookScheduler.css:186]: @Jeyaprakash-NK Should background-color be a variable or is this fine? Please check once.

**Review by harsha-accenture — CHANGES_REQUESTED** [review]

File: src/scheduler/createNotebookScheduler.tsx [hunk 3]
```diff
@@ -0,0 +1,823 @@
...
+import { Cron } from 'react-js-cron';
+import 'react-js-cron/dist/styles.css';
+import { KernelSpecAPI } from '@jupyterlab/services';
+import tzdata from 'tzdata';
+import { SchedulerService } from './schedulerServices';
+import NotebookJobComponent from './notebookJobs';
+import LeftArrowIcon from '../../style/icons/left_arrow_icon.svg';
+import { LabIcon } from '@jupyterlab/ui-components';
+import errorIcon from '../../style/icons/error_icon.svg';
+// import EditIcon from '../../style/icons/edit_icon_disable.svg';
```
[the remark this answers is not in the document]
Comment (Jeyaprakash-NK) [reply src/scheduler/createNotebookScheduler.tsx]: resolved - removed this line

File: src/scheduler/createNotebookScheduler.tsx [hunk 4]
```diff
@@ -0,0 +1,823 @@
...
+const iconLeftArrow = new LabIcon({
+  name: 'launcher:left-arrow-icon',
+  svgstr: LeftArrowIcon
+});
+
+const iconError = new LabIcon({
+  name: 'launcher:error-icon',
+  svgstr: errorIcon
+});
+// const iconEdit = new LabIcon({
```
[the remark this answers is not in the document]
Comment (Jeyaprakash-NK) [reply src/scheduler/createNotebookScheduler.tsx]: resolved - removed this line

File: src/scheduler/createNotebookScheduler.tsx [hunk 5]
```diff
@@ -0,0 +1,823 @@
...
+}: {
+  themeManager: IThemeManager;
+  app: JupyterLab;
+  context: any;
+}): JSX.Element => {
+  const [jobNameSelected, setJobNameSelected] = useState('');
+  const [inputFileSelected, setInputFileSelected] = useState('');
+  const [composerList, setComposerList] = useState<string[]>([]);
+  const [composerSelected, setComposerSelected] = useState('');
+  // const [outputNotebook, setOutputNotebook] = useState(true);
```
[the remark this answers is not in the document]
Comment (Jeyaprakash-NK) [reply src/scheduler/createNotebookScheduler.tsx]: resolved - removed this line

File: src/scheduler/createNotebookScheduler.tsx [hunk 6]
```diff
@@ -0,0 +1,823 @@
...
+      setDagListCall(false)
+      return true;
+    } catch (error) {
+      setDagListCall(false)
+      console.error('Error checking job name uniqueness:', error);
+      return false;
+    }
+  };
+
+  // const handleOutputNotebookChange = (
```
[the remark this answers is not in the document]
Comment (Jeyaprakash-NK) [reply src/scheduler/createNotebookScheduler.tsx]: resolved - removed this line

File: src/scheduler/createNotebookScheduler.tsx [hunk 7]
```diff
@@ -0,0 +1,823 @@
...
+    setEmailOnSuccess(event.target.checked);
+  };
+
+  const handleEmailList = (data: string[]) => {
+    setEmailList(data);
+  };
+
+  const handleCreateJobScheduler = async () => {
+    let outputFormats = [];
+    // if (outputNotebook) {
```
[the remark this answers is not in the document]
Comment (Jeyaprakash-NK) [reply src/scheduler/createNotebookScheduler.tsx]: resolved - removed this line

File: src/scheduler/createNotebookScheduler.tsx [hunk 8]
```diff
@@ -0,0 +1,823 @@
...
+    outputFormats.push('ipynb');
+    // }
+    // if (outputHtml) {
```
[the remark this answers is not in the document]
Comment (Jeyaprakash-NK) [reply src/scheduler/createNotebookScheduler.tsx]: resolved - removed this line

File: src/scheduler/createNotebookScheduler.tsx [hunk 9]
```diff
@@ -0,0 +1,823 @@
...
+    );
+    setEditMode(false);
+  };
+
+  const handleJobNameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    event.target.value.length > 0
+      ? setJobNameValidation(true)
+      : setJobNameValidation(false);
+
+    const regexp = /^[a-zA-Z0-9-_]+$/;
```
[the remark this answers is not in the document]
Comment (Jeyaprakash-NK) [reply src/scheduler/createNotebookScheduler.tsx:276]: resolved - Added comment for regex

File: src/scheduler/createNotebookScheduler.tsx [hunk 10]
```diff
@@ -0,0 +1,823 @@
...
+      } else {
+        const selectedData: any = clusterList.filter((cluster: string) => {
+          return context.sessionContext.kernelDisplayName.includes(cluster);
+        });
+        setClusterSelected(selectedData[0]);
+      }
+    }
+  };
+
+  // const handleEditNotebookData = async () => {
```
[the remark this answers is not in the document]
Comment (Jeyaprakash-NK) [reply src/scheduler/createNotebookScheduler.tsx]: resolved - removed this line

File: src/scheduler/createNotebookScheduler.tsx [hunk 11]
```diff
@@ -0,0 +1,823 @@
...
+            )}
+
+            <div className="create-scheduler-form-element-input-file">
+              <Input
+                className="create-scheduler-style"
+                value={inputFileSelected}
+                Label="Input file*"
+                disabled={true}
+              />
+              {/* {inputFileSelected.includes('gs://') && (
```
[the remark this answers is not in the document]
Comment (Jeyaprakash-NK) [reply src/scheduler/createNotebookScheduler.tsx]: resolved - removed this line

File: src/scheduler/createNotebookScheduler.tsx [hunk 12]
```diff
@@ -0,0 +1,823 @@
...
+            <div className="create-scheduler-form-element">
+              <FormGroup row={true}>
+                <FormControlLabel
+                  control={
+                    <Checkbox
+                      size="small"
+                      readOnly
+                      checked={true}
+                      defaultChecked={true}
+                      // onChange={handleOutputNotebookChange}
```
[the remark this answers is not in the document]
Comment (Jeyaprakash-NK) [reply src/scheduler/createNotebookScheduler.tsx]: resolved - removed this line

File: src/scheduler/createNotebookScheduler.tsx [hunk 13]
```diff
@@ -0,0 +1,823 @@
...
+                    />
+                  }
+                  className="create-scheduler-label-style"
+                  label={
+                    <Typography sx={{ fontSize: 13 }}>Notebook</Typography>
+                  }
+                />
+                {/* <FormControlLabel
```
[the remark this answers is not in the document]
Comment (Jeyaprakash-NK) [reply src/scheduler/createNotebookScheduler.tsx]: resolved - removed this line

File: src/scheduler/executionHistory.tsx [hunk 14]
```diff
@@ -0,0 +1,218 @@
...
+  const [blueListDates, setBlueListDates] = useState<string[]>([]);
+  const [greyListDates, setGreyListDates] = useState<string[]>([]);
+  const [orangeListDates, setOrangeListDates] = useState<string[]>([]);
+  const [redListDates, setRedListDates] = useState<string[]>([]);
+  const [greenListDates, setGreenListDates] = useState<string[]>([]);
+  const [darkGreenListDates, setDarkGreenListDates] = useState<string[]>([]);
+
+  const handleMonthChange = () => {
+    setDagRunId('');
+    // setStartDate('');
```
[the remark this answers is not in the document]
Comment (Jeyaprakash-NK) [reply src/scheduler/executionHistory.tsx]: resolved - removed this line

File: src/scheduler/executionHistory.tsx [hunk 15]
```diff
@@ -0,0 +1,218 @@
...
+    // setEndDate('');
+    setSelectedDate(null);
+  };
+
+  const handleDateSelection = (selectedValue: any) => {
+    setDagRunId('');
+    // setStartDate('');
```
[the remark this answers is not in the document]
Comment (Jeyaprakash-NK) [reply src/scheduler/executionHistory.tsx]: resolved - removed this line

File: src/scheduler/executionHistory.tsx [hunk 16]
```diff
@@ -0,0 +1,218 @@
...
+    }
+    if (isLastVisibleCell) {
+      const nextDate = new Date(day.toDate());
+      nextDate.setDate(day.toDate().getDate() + 1);
+      setEndDate(nextDate.toISOString());
+    }
+
+    const totalViewDates = day.date();
+
+    const isBlueExecution =
```
[the remark this answers is not in the document]
Comment (Jeyaprakash-NK) [reply src/scheduler/executionHistory.tsx:81]: resolved - Added comment for color code details

File: src/scheduler/executionHistory.tsx [hunk 17]
```diff
@@ -0,0 +1,218 @@
...
+          </div>
+          <div className="create-job-scheduler-title">
+            Execution History: {dagId} 
+          </div>
+        </div>
+        <div className="execution-history-main-wrapper">
+          <div className="execution-history-left-wrapper">
+            <LocalizationProvider dateAdapter={AdapterDayjs}>
+              <DateCalendar
+                // value={selectedDate}
```
[the remark this answers is not in the document]
Comment (Jeyaprakash-NK) [reply src/scheduler/executionHistory.tsx]: resolved - removed this line

File: src/scheduler/listDagRuns.tsx [hunk 18]
```diff
@@ -0,0 +1,245 @@
...
+    );
+  };
+
+  const listDagRunsList = async () => {
+    await SchedulerService.listDagRunsListService(
+      composerName,
+      dagId,
+      startDate,
+      endDate,
+      // selectedDate,
```
[the remark this answers is not in the document]
Comment (Jeyaprakash-NK) [reply src/scheduler/listDagRuns.tsx]: resolved - removed this line

File: src/scheduler/schedulerServices.tsx [hunk 19]
```diff
@@ -0,0 +1,683 @@
...
+      setEditDagLoading('');
+    }
+  };
+
+  static listDagRunsListService = async (
+    composerName: string,
+    dagId: string,
+    startDate: string,
+    endDate: string,
+    // selectedDate: Dayjs | null,
```
[the remark this answers is not in the document]
Comment (Jeyaprakash-NK) [reply src/scheduler/schedulerServices.tsx]: resolved - removed this line

File: src/scheduler/schedulerServices.tsx [hunk 20]
```diff
@@ -0,0 +1,683 @@
...
+    setGreyListDates: (value: string[]) => void,
+    setOrangeListDates: (value: string[]) => void,
+    setRedListDates: (value: string[]) => void,
+    setGreenListDates: (value: string[]) => void,
+    setDarkGreenListDates: (value: string[]) => void
+  ) => {
+    setIsLoading(true);
+    let start_date = startDate;
+    let end_date = endDate;
+    // if (selectedDate !== null) {
```
[the remark this answers is not in the document]
Comment (Jeyaprakash-NK) [reply src/scheduler/schedulerServices.tsx]: resolved - removed this line

File: src/scheduler/schedulerServices.tsx (hunk 1 above)
Comment (Jeyaprakash-NK) [reply src/scheduler/schedulerServices.tsx]: resolved - removed console log

File: style/notebookScheduler.css (hunk 2 above)
Comment (Jeyaprakash-NK) [reply style/notebookScheduler.css:186]: resolved - discussed about this css color code issue

**Review by harsha-accenture — APPROVED** [review]

**Pull request merged**